Assets/Scripts/Environment/Turret.cs

```diff
@@ -0,0 +1,99 @@
+    // Prefab for projectile that will be shot from the turret.
+    [SerializeField] private GameObject _projectilePrefab;
+    // Location for projectiles to be spawned from, should be a location in front of the turret.
+    [SerializeField] private GameObject _projectileSpawn;
+    // An audio source that provides a "detection" sound for the turret.
+    private AudioSource _audio;
+
+    void Awake()
+    {
+        _health = GetComponent<HealthSystem>();
+        _audio = GetComponent<AudioSource>();
+    }
+
+    void Update()
+    {
+        // If the turret's health is 0, make it explode and destroy itself.
+        if (_health.CurrentHealth <= 0)
+        {
+            Die();
+            return;
+        }
+        // If turret is ready to check for targets, reset its time towards the next check and try to find a target.
+        if (_timeToNextTargetCheck >= _targetCheckTime)
+        {
+            _timeToNextTargetCheck = 0;
+            GameObject player = FindTarget();
+            // If a player is found and turret is not on cooldown, shoot at the player and reset cooldown progress.
+            if (player != null  && _cooldownProgress >= _attackCooldown)
+            {
+                Shoot(player);
+                _cooldownProgress = 0;
+            }
+        }
+        // Update the cooldown of the turret.
+        if (_cooldownProgress < _attackCooldown)
+        {
+            _cooldownProgress += Time.deltaTime;
+        }
+        // Progress time towards the next target check.
+        _timeToNextTargetCheck += Time.deltaTime;
+    }
+
+    private void Shoot(GameObject target)
+    {
+        // Create an energy projectile and initialize it with a target position.
+        GameObject projectileObj = Instantiate(_projectilePrefab, _projectileSpawn.transform.position, Quaternion.identity);
+        EnergyProjectile projectile = projectileObj.GetComponent<EnergyProjectile>();
+        projectile.InitializeWithTargetPos(target.transform.position);
+    }
+
+    private void Die()
+    {
+        // Create an explosion at the turret's location and destroy itself.
+        Instantiate(_explosionPrefab, transform.position, Quaternion.identity);
+        Destroy(gameObject);
+    }
+
+    private GameObject FindTarget()
+    {
+        Collider[] collidersInRange = Physics.OverlapSphere(transform.position, _detectionRadius);
+        // Find a target within detection range of the turret.
+        foreach (Collider collider in collidersInRange)
+        {
+            bool isValidTarget = collider.CompareTag("Player");
+            // If a collider is a valid target, look at it and play a beeping detection sound.
+            if (isValidTarget)
+            {
+                if (!_audio.isPlaying)
+                {
+                    _audio.Play();
+                }
+                transform.LookAt(collider.transform);
+                return collider.gameObject;
+            }
+        }
+        _audio.Stop();
```

Comment:
Won`t this stop the audio from playing once you find the target? Seems you're trying to play audio when you find a target which is cool.

Reply:
The turret should play a "detection" audio as long as a player is in its range. By stopping the audio source, the turret doesn't keep playing the audio even after the player is out of range.